skrub/_joiner.py

```diff
@@ -155,30 +144,23 @@ def fit(self, X: pd.DataFrame, y=None) -> "Joiner":
         Joiner
             Fitted Joiner instance (self).
         """
+        self._main_key, self._aux_key = _join_utils.check_key(
+            self.main_key, self.aux_key, self.key
+        )
 
-        main_key_list = np.atleast_1d(self.main_key).tolist()
-
-        for col in main_key_list:
+        for col in self._main_key:
             if col not in X.columns:
                 raise ValueError(
                     f"Main key {col!r} not found in columns of X:"
                     f" {X.columns.tolist()}. "
                 )
 
-        if isinstance(self.tables[0], tuple):
-            self.tables_ = self.tables
-        else:
-            self.tables_ = list()
-            self.tables_.append(tuple(self.tables))
-
-        for table_idx, (df, cols) in enumerate(self.tables_):
-            cols = np.atleast_1d(cols).tolist()
-            for col in cols:
-                if col not in df.columns:
-                    raise ValueError(
-                        f"Column key {col!r} not found in columns of "
-                        f"table index {table_idx}: {df.columns.tolist()}. "
-                    )
+        for col in self._aux_key:
+            if col not in self.aux_table.columns:
+                raise ValueError(
+                    f"Column key {col!r} not found in columns of "
+                    f"auxiliary table: {self.aux_table.columns.tolist()}. "
+                )
```

Comment:
This would not make sense to have it in `check_key`? Or is it really specific to this class?

Reply:
Actually the `check_key` expect `aux_key`, `main_key` so I would expect to make such check as well.

Reply:
actually I'm not sure this check is super useful, if we don't do it we get a more accurate KeyError when trying to use the key

Reply:
So maybe we should think of removing it then.

Reply:
It's nice to raise an error early on (during `fit` instead of during `predict`), IMHO. I'm +1 on keeping it or putting it into `check_key` as @glemaitre suggests since both checks will be performed in practice.

Reply:
I put it in a different function as check_key is rather for figuring out what key to use from main_key, aux_key and key, and because this check happens in fit and transform whereas check_key is only in fit